api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -478,16 +478,33 @@ message HttpConnectionManager {
   // Should paths be normalized according to RFC 3986 before any processing of
   // requests by HTTP filters or routing? This affects the upstream *:path* header
   // as well. For paths that fail this check, Envoy will respond with 400 to
-  // paths that are malformed. This defaults to false currently but will default
-  // true in the future. When not specified, this value may be overridden by the
-  // runtime variable
+  // paths that are malformed. This defaults to true.
+  // This by default affects the upstream *:path* header as well.
+  // When not specified, this value may be overridden by the runtime variable
   // :ref:`http_connection_manager.normalize_path<config_http_conn_man_runtime_normalize_path>`.
+  // The upstream path normalization can be specified independently, see
+  // :ref:`normalize_path_for_upstream
+  // <envoy_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.normalize_path_for_upstream>`.
   // See `Normalization and Comparison <https://tools.ietf.org/html/rfc3986#section-6>`
   // for details of normalization.
   // Note that Envoy does not perform
   // `case normalization <https://tools.ietf.org/html/rfc3986#section-6.2.2.1>`
   google.protobuf.BoolValue normalize_path = 30;
 
+  // Should paths to upstream be normalized according to RFC 3986 after any processing of
```

Comment:
A few questions to get started:
1) When does this return 400 and if we want to return a 400 why would someone do this here vs. just doing it at the beginning before filters run? It's not clear to me what utility this has.
2) Should we actually deprecate normalize_path and have a new field that is an enum that describes which normalizing approach to use? Also, in the above docs it says that normalize_path defaults to true. I thought it defaults to false or maybe I am confused?

Reply:
1.  Agreed, yes that makes sense.
2.  The addition on a the extra header was intended to minimize the impact on routing/filtering code changes to deal with potential changes to, as you say, replacing normalize_path with an enum.  Idealy yes an enum is a cleaner approach.

Reply:
A deprecation makes sense IMHO, as long as we preserve existing behavior today.